packages/blocks/src/list-block/list-block.ts

```diff
@@ -90,98 +89,38 @@ export class ListBlockComponent extends NonShadowLitElement {
+      this.host.page.updateBlock(this.model, checkedPropObj);
+      return;
+    }
+    selectList(this.model);
+  };
 
   firstUpdated() {
-    // this.pageAwarenessStore.awareness.on('change', () => {
-    //   this.requestUpdate();
-    // });
     this.model.propsUpdated.on(() => this.requestUpdate());
-    this.model.childrenUpdated.on(() => {
-      if (this.isToggleEnabled) {
-        if (!this.model.children.length) {
-          this.toggleHiddenChildren('hide'); // if there are no children set to hide (so we get a closed disabled arrow)
-        } else {
-          this.toggleHiddenChildren('show');
-        }
-      } else {
-        this.requestUpdate();
-      }
-    });
+    this.model.childrenUpdated.on(() => this.requestUpdate());
   }
-  toggleHiddenChildren = (overRide?: 'show' | 'hide') => {
-    if (!this.isToggleEnabled) return;
-
-    const currentBlockID = this.model.id;
-    const hiddenBlockListWithoutThis =
-      this.blocksWithHiddenChildrenOnPage.filter(
-        (eachBlockID: string) => eachBlockID !== currentBlockID
-      );
-
-    if (
-      !(overRide === 'hide') &&
-      this.hasChildren &&
-      (overRide === 'show' || this.hasHiddenChildren)
-    ) {
-      this.model.page.blocksWithHiddenChildren = hiddenBlockListWithoutThis; // remove current block from the hiddenChildren list
-    } else {
-      this.model.page.blocksWithHiddenChildren = [
-        ...hiddenBlockListWithoutThis,
-        currentBlockID,
-      ]; // ensure current block is in the hiddenChildren list
-
-      // attempt to avoid mouse move / selection errors
-      // for (const eachChildBlock of this.model.children) {
-      //   eachChildBlock.selection.dispose();
-      // }
-    }
-    asyncFocusRichText(this.model.page, this.model.id);
-    this.requestUpdate();
-  };
 
   render() {
     this.setAttribute(BLOCK_ID_ATTR, this.model.id);
     const { deep, index } = getListInfo(this.host, this.model);
-    const listIcon = getListIcon({
-      model: this.model,
-      deep,
-      index,
-      onClick: evt => {
-        // evt.preventDefault();
-        // evt.stopPropagation();
-        if (this.model.type === 'toggle') {
-          // this.host.page.captureSync();
-          this.toggleHiddenChildren();
-          return;
-        } else if (this.model.type === 'todo') {
-          this.host.page.captureSync();
-          const checkedPropObj = { checked: !this.model.checked };
-          this.host.page.updateBlock(this.model, checkedPropObj);
-          return;
-        }
-        selectList(this.model);
-      },
-    });
-    const childrenContainer =
-      this.model.type === 'toggle' && this.hasHiddenChildren
-        ? null
-        : BlockChildrenContainer(this.model, this.host, () =>
-            this.requestUpdate()
-          );
+    const { model, showChildren, _onClickIcon } = this;
+    const listIcon = ListIcon(model, deep, index, showChildren, _onClickIcon);
+
+    const childrenContainer = this.showChildren
```

Comment:
yeah, i thought about this strategy, but kept the prop at the page level, 
as I predict other ways of hiding children of other types will be eventually desirable.

And still, i get that it feels tighter this way.

Thanks for working on Sunday to continue the PR!